src/modules/HttpServerPackages.rb

```diff
@@ -34,13 +34,13 @@ def main
     #
     # @param pattern [String] a regex pattern to match, no escaping done
     # @return list of matching package names
-    def by_pattern(pattern)
+    def by_provides_regexp(pattern)
       raise ArgumentError, "Missing search pattern" if pattern.nil? || pattern.empty?
 
       # NOTE: - Resolvable.find takes POSIX regexp, later select uses Ruby regexp
       # - Resolvable.find supports regexps only for dependencies, so we need to
       # filter result according to package name
-      Y2Packager::Resolvable.find({ provides_regexp: "^#{pattern}$" }, [:name])
+      Y2Packager::Resolvable.find({ kind: :package, provides_regexp: "^#{pattern}$" }, [])
```

Comment:
More simplification:
```suggestion
      Y2Packager::Resolvable.find(kind: :package, provides_regexp: "^#{pattern}$")
```